### ISSUE TITLE
Indiscriminate Weapon attribute

### ISSUE THREAD

**Issue by pilover100** [endless-sky/endless-sky#7891]
### Problem Description

Currently weapons fire completely ignores allied ships (except for the blast of a non-safe AoE weapon like heavy rockets when they get to their target or hit an asteroid).

With almost complete control of the projectile's trajectory that's fine, but for weapons such as flak cannons or other weapons that produce unreliable cones of projectiles (such as the new-ish korath Heaver) it would not be possible to easily avoid allies.

What I propose is adding an attribute to the weapon that if present forces the projectile to collide with and damage any ship it hits regardless of hostilities.

### Related Issue Links

No related issues I have found, though I could easily have missed things here and there

### Desired Solution

I would desire that a tag such as `indiscriminate` be added that causes projectiles to damage any ship they collide with regardless of government, similar to the `cluster` and `stream` tags that toggle firing behaviour.

Addition of this would add more flavour to a weapon and could lead to more diversity of weapons depending on your situation.

### Alternative Approaches

Blast radius could be used for projectiles such as bombs, but would make flak fields more unwieldy as they would still potentially move through allied ships without damaging them.

**Comment by vitalchip** [comment]
I've tried this in the past, I was looking for a way to make the flamethrower acceptable (I ended up just removing it).

What I'd tried was adding a "hitsAll" attribute to weapons which did what it says, even your own ship. Then submunitions from the flames would damage your own ship if you flew into them or put the flamethrower in a stupid place on the ship etc.

The basic principle is simple to add but it raises issues around provoking other ships, particularly from escorts as they currently shoot on the assumption the shots only hit enemies. There's a similar issue with flak barriers.

It's something to bear in mind because provoking an otherwise neutral faction might have long lasting consequences.

Ships would need to check what's in their line of fire beforehand and other ships would have to actually start looking where they're going because if they fly right into your engagement zone and get hit it'll still be your 'fault'.

**Comment by Zitchas** [comment]
*nod* yeah, that does add a bit of complexity requirements to the AI.

The EV series had that, and it definitely made some battles challenging as one had to keep holding fire to avoid shooting "brave merchants" who were also trying to engage the target, or stuff like that. Although I think EV battles tended to be smaller and fewer deathballs, so hitting other people targeting the same ship was less of an issue. That being said, I remember having to run away from merchants or other normally-neutral ships because I'd provoked them and didn't ant the rep hit from killing them.

**Comment by vitalchip** [comment]
There are all sorts of possibilities but what tends to happen in my experience is that you need a series of ideas to come together to get a larger idea to work and it's enough bother trying to get one PR through without it being contingent on the next five.

It's the sort of thing a single developer or a 'led team' can achieve but a community based project has all but no chance.

The ai firing could check for friendly targets that were either in the line of fire now or heading into the line of fire, not going to catch every situation but if we also made provoking temporary or forbearing when hit by friendly fire which didn't haave that ship as the target. Still causes damage, still could swing a battle but not going to lock you out of any missions.

 Some things would be popular though for potentially undesirable reasons, if the ai has a target and there's a disabled ship in the way I'd be looking for it to hold fire, re=position and then take the shot..  more disabled ships saved = more capturing which in my view is a good behaviour creating a bad situation.

**Comment by Zitchas** [comment]
Yeah. This was one of the reasons that the Revamp PR was so popular, I think. Because it slammed together what, about a dozen active PRs and a whole slew of arbitrary changes that I thought were beneficial... And it worked! Mostly. It had a lot of adjustments, and had some stuff that was broken. Definitely wasn't a polished thing suitable for merging. But it did demonstrate that a lot of these concepts that were totally dismissed out of hand as being deleterious actually had a positive impact on the game when done alongside other things. Some things actually got merged and implemented in vanilla as a result of it. Some things... are still waiting. Poor formations. Most people loved them, everyone thought it made the game look really good. But there were, alas, technical hangups. The fighter behavior was also exemplary. Between giving them mobility to match the popculture view of fighters and an AI that can take advantage of that... A pack of fighters was actually kind of terrifying. Carrying anti-fighter stuff suddenly had all sorts of appeal and use.

**Comment by johndh** [comment]
Friendly fire sounds like an incredibly frustrating thing to introduce.
